app/models/ring.rb

```diff
@@ -1,19 +1,18 @@
 class Ring
   def initialize(redirection_to_link)
-    @oldest_redirection = Redirection.order(created_at: :desc).last
-    @newest_redirection = Redirection.order(created_at: :desc).first
     @redirection_to_link = redirection_to_link
+    @from_redirection = Redirection.first
+    @to_redirection = from_redirection.next
```

Comment:
Actually, that code snippet makes everything break, probably because it's grabbing `Redirection.first.next` while we're fiddling with the ring. We need to grab `Redirection.first.next` before doing anything fancy, i.e. in the constructor.